xmlhttprequest.php

```diff
@@ -43,6 +43,8 @@
 	log_event( LOG_AJAX, 'DEPRECATED: Calling {' . $t_function . '}. Use REST API instead.' );
 	call_user_func( $t_function );
 } else {
-	log_event( LOG_AJAX, 'Unknown function for entry point = ' . $t_function );
-	echo 'unknown entry point';
+	$t_msg = 'Unknown function for entry point: ' . $t_function;
+	log_event( LOG_AJAX, $t_msg );
+	header( 'HTTP/1.1 500 ' . $t_msg, false, 500 );
+	echo $t_msg;
```

Comment:
If I try to inject a CR/LF in the header, PHP returns _Warning: Header may not contain more than a single header, new line detected_ so I think it's safe. Confirmed by [PHP documentation](http://php.net/manual/en/function.header.php#refsect1-function.header-changelog).